src/ClientConfig.cc

```diff
@@ -340,7 +350,12 @@ bool ClientConfig::LoadConfig(const std::string &_file)
               // Add the new server.
               ServerConfig newServer;
               newServer.SetUrl(common::URI(serverURL));
-              this->AddServer(newServer);
```

Comment:
Don't we still need to add the new server? Otherwise it looks like it will be destroyed at the end of this scope.